docs/source/cli-reference/basic-options.rst

```diff
@@ -0,0 +1,239 @@
+    - 50 - Default.
+
+    An example usage::
+
+        scancode -clpieu --json-pp output.json samples --max-email 5
+
+    This only reports 5 email addresses per file and ignores the rest.
+
+----
+
+``--max-url`` Options
+---------------------
+
+    .. admonition:: Dependency
+
+        The option ``--max-url`` is a sub-option of and requires the option ``--url``.
+
+    If in the files that are scanned, in individual files, there are a lot of links to other websites
+    (i.e url lists) which are unnecessary and clutter the scan results, ``--max-url`` option can be
+    used to report urls only up to a limit in individual files.
+
+    Some important INTEGER values of the ``--max-url INTEGER`` option:
+
+    - 0  - No limit, include all urls.
+    - 50 - Default.
+
+    An example usage::
+
+        scancode -clpieu --json-pp output.json samples --max-url 10
+
+    This only reports 10 urls per file and ignores the rest.
+
+----
+
+``--license-score`` Options
+---------------------------
+
+    .. admonition:: Dependency
+
+        The option ``--license-score`` is a sub-option of and requires the option ``--license``.
+
+    ..
+        [ToDo] Research and Write License Matching Better
+
+    License matching strictness, i.e. How closely matched licenses are detected in a scan, can be
+    modified by using this ``--license-score`` option.
+
+    Some important INTEGER values of the ``--license-score INTEGER`` option:
+
+    - **0**     - Default and Lowest Value, All matches are reported.
+    - **100**    - Highest Value, Only licenses with a much better match are reported
+
+    Here, a bigger number means a better match, i.e. Setting a higher license score translates to a
+    higher threshold for matching licenses (with equal or less number of license matches).
+
+    An example usage::
+
+        scancode -clpieu --json-pp output.json samples --license-score 70
+
+    Here's the license results on setting the integer value to 100, Vs. the default value 0. This is
+    visualized using ScanCode workbench in the License Info Dashboard.
+
+    .. list-table:: License scan results of Samples Directory.
+
+        * - .. figure:: data/core_lic_score_0.png
+
+               License Score 0 (Default).
+
+          - .. figure:: data/core_lic_score_100.png
+
+               License Score 100.
+
+----
+
+``--license-text`` Options
+--------------------------
+
+    .. admonition:: Dependency
+
+        The option ``--license-text`` is a sub-option of and requires the option ``--license``.
+
+    .. admonition:: Sub-Option
+
+        The option ``--license-text-diagnostics`` and ``--is-license-text`` are sub-options of
+        ``--license-text``. ``--is-license-text`` is a Post-Scan Option.
+
+    With the ``--license-text`` option, the scan results attribute "matched text" includes the matched text
+    for the detected license.
+
+    An example Scan::
+
+        scancode -cplieu --json-pp output.json samples --license-text
```

Comment:
I think for the basic options documentation, we need to focus on these particular options in this command. 

For example, we should restructure this doc to:
1) tell the user what each of the `-cplieu` flags mean. These are the primary options that people are interested in using
2) tell the user what `--json-pp` means, AND some of the various other output formats like `csv`
3) Move the rest of the option docs to `core-options.rst` OR `extra-options.rst` or similar. 

Basically, we want to have `basic-options.rst` to cover just the basic options that a new user would want to know about before running scancode. More advanced users can look elsewhere for more information.

Reply:
I'm looking into point `1`.  

And IMO as output-format.rst already has things mentioned in `2`, should I add links to it? 

So what I was trying to achieve through this page structure is grouping command-line options consistent with that of the help text (and groups used while coding). So moving other options (other than `-cplieu`) to another page is still okay?

Since your other comments also raise similar concerns with the grouping of CLI options, I'm starting a discussion @ gitter, in the gsod channel.

Reply:
Basically what I mean is, `synopsis.rst` is the page that gives basic users an overview of the options, outputs, and scan command variation. `basic-options.rst` just has the CLI options grouped as basic options.

Reply:
Yes, I suppose following the various groups is fine. I had forgotten that this is how they are structured internally in the code itself.